lib/pure/algorithm.nim

```diff
@@ -559,6 +561,85 @@ proc isSorted*[T](a: openArray[T], order = SortOrder.Ascending): bool =
     assert isSorted(e) == false
   isSorted(a, system.cmp[T], order)
 
+proc merge*[T](
+  result: var seq[T],
+  x, y: openArray[T], cmp: proc(x, y: T): int {.closure.}
+) {.since: (1, 5, 1).} =
+  ## Merges two sorted `openArray`. `x` and `y` are assumed to be sorted.
+  ## If you do not wish to provide your own `cmp`,
+  ## you may use `system.cmp` or instead call the overloaded
+  ## version of `merge`, which uses `system.cmp`.
+  ##
+  ## **See also:**
+  ## * `merge proc<#merge,seq[T],openArray[T],openArray[T]>`_
+  runnableExamples:
+    let x = @[1, 3, 6]
+    let y = @[2, 3, 4]
+
+    var merged: seq[int]
+    merged.merge(x, y, system.cmp[int])
+    assert merged.isSorted
+    assert merged == @[1, 2, 3, 3, 4, 6]
+
+    import sugar
+
+    var res: seq[(int, int)]
+    res.merge([(1,1)], [(1,2)], (a,b) => a[0] - b[0])
+    assert res == @[(1, 1), (1, 2)]
+
+  let
+    sizeX = x.len
+    sizeY = y.len
+
+  result.setlen(sizeX + sizeY)
+
+  var
+    ix = 0
+    iy = 0
+    i = 0
+
+  while true:
+    if ix == sizeX:
+      while iy < sizeY:
+        result[i] = y[iy]
+        inc i
+        inc iy
+      return
+
+    if iy == sizeY:
+      while ix < sizeX:
+        result[i] = x[ix]
+        inc i
+        inc ix
+      return
+
+    let itemX = x[ix]
+    let itemY = y[iy]
+
+    if cmp(itemX, itemY) > 0:
+      result[i] = itemY
+      inc iy
+    else:
+      result[i] = itemX
+      inc ix
+
+    inc i
+
+proc merge*[T](result: var seq[T], x, y: openArray[T]) {.since: (1, 5, 1).} =
```

Comment:
```suggestion
proc merge*[T](result: var seq[T], x, y: openArray[T]) {.inline, since: (1, 5, 1).} =
```

(otherwise, inlining cannot happen unless caller is in same module or --passc:-flto is used)

Reply:
done